tests/store/tracking/test_sqlalchemy_store.py

```diff
@@ -7404,3 +7404,347 @@ def test_scorer_operations(store: SqlAlchemyStore):
+        request_time=timestamp_ms,
+        execution_duration=100,
+        state=TraceState.OK,
+        tags=tags or {},
+    )
+
+    return store.start_trace(trace_info)
+
+
+def _create_trace_for_correlation(store, experiment_id, spans=None, assessments=None, tags=None):
+    """
+    Simplified trace creation for correlation tests.
+
+    Since correlation tests only filter on tags, we don't actually need to create real spans.
+    We just set the appropriate tags based on what spans would have been created.
+    """
+    trace_id = f"tr-{uuid.uuid4()}"
+    timestamp_ms = time.time_ns() // 1_000_000
+
+    trace_tags = tags or {}
+
+    # If spans are specified, set tags based on them (for backward compatibility)
+    if spans:
+        span_types = [span.get("type", "LLM") for span in spans]
+        span_statuses = [span.get("status", "OK") for span in spans]
+
+        # Set primary span type
+        if "TOOL" in span_types:
+            trace_tags["primary_span_type"] = "TOOL"
+        elif "LLM" in span_types:
+            trace_tags["primary_span_type"] = "LLM"
+
+        # Set type flags
+        if "LLM" in span_types:
+            trace_tags["has_llm"] = "true"
+        if "TOOL" in span_types:
+            trace_tags["has_tool"] = "true"
+
+        # Set error flag
+        trace_tags["has_error"] = "true" if "ERROR" in span_statuses else "false"
+
+        # Count tools
+        tool_count = sum(1 for t in span_types if t == "TOOL")
+        if tool_count > 0:
+            trace_tags["tool_count"] = str(tool_count)
+
+    trace_info = TraceInfo(
+        trace_id=trace_id,
+        trace_location=trace_location.TraceLocation.from_experiment_id(experiment_id),
+        request_time=timestamp_ms,
+        execution_duration=100,
+        state=TraceState.OK,
+        tags=trace_tags,
+    )
+    store.start_trace(trace_info)
+
+    if assessments:
+        for assessment_data in assessments:
+            assessment = Feedback(
+                assessment_id=assessment_data.get("assessment_id", f"fb-{uuid.uuid4()}"),
+                trace_id=trace_id,
+                name=assessment_data.get("name", "quality"),
+                assessment_type=assessment_data.get("assessment_type", "feedback"),
+                source=AssessmentSource(
+                    source_type=AssessmentSourceType.HUMAN,
+                    source_id=assessment_data.get("source_id", "user123"),
+                ),
+                value=FeedbackValue(assessment_data.get("value", 0.8)),
+                created_timestamp=timestamp_ms,
+                last_updated_timestamp=timestamp_ms,
+            )
+            store.log_assessments([assessment])
+
+    return trace_id
+
+
+def _create_trace_with_spans_for_correlation(store, experiment_id, span_configs):
+    return _create_trace_for_correlation(store, experiment_id, spans=span_configs)
+
+
+def test_calculate_trace_filter_correlation_basic(store):
```

Comment:
@BenWilson2 Can we add a test case for base filter and verify that the NPMI result fields match the expected values when the base filter is applied? I.e. the total should be the total number of traces matching the base filter.